src/internal/connector/graph/service.go

```diff
@@ -160,6 +172,40 @@ func HTTPClient(opts ...option) *http.Client {
 	return httpClient
 }
 
+// GetDefaultMiddlewares creates a new default set of middlewares for the Kiota request adapter
+func GetMiddlewares() []khttp.Middleware {
+	return []khttp.Middleware{
+		// khttp.NewRetryHandler(),
+		khttp.NewRedirectHandler(),
+		khttp.NewCompressionHandler(),
+		khttp.NewParametersNameDecodingHandler(),
+		khttp.NewUserAgentHandler(),
+		&LoggingMiddleware{},
+		&RetryHandler{
```

Comment:
As per the meeting, our retry handler should be last to occur in the middleware stack (ie, idx 0), not first.